# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoaderBase.cpp
# Commit: 997e9ca2e1e9
# Full Hash: 997e9ca2e1e9d636f8f956151db81ca2a8a60c82
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement InitModuleLoader methods for WorkerScriptLoader and WorkerGlobalScope; r=jonco
#   
#   This implements a method to initialize the moduleLoader for workers. This will initialize only once, for all worker types (module and classic).
#   
#   Depends on D147324
# ==============================================================================

diff -r ebeb65b42212 -r 997e9ca2e1e9 js/loader/ModuleLoaderBase.cpp
--- a/js/loader/ModuleLoaderBase.cpp	Wed Jan 18 13:46:29 2023 +0000
+++ b/js/loader/ModuleLoaderBase.cpp	Wed Jan 18 13:46:29 2023 +0000
@@ -982,7 +982,9 @@
   MOZ_ASSERT(mFetchingModules.IsEmpty());
 
   for (const auto& entry : mFetchedModules) {
-    entry.GetData()->Shutdown();
+    if (entry.GetData()) {
+      entry.GetData()->Shutdown();
+    }
   }
 
   mFetchedModules.Clear();
